call-home/src/collector/report_models.rs

```diff
@@ -204,65 +183,62 @@ impl Report
 }
 
 /// Get maximum value from a vector
-fn get_max_value(values: Vec<u64>) -> u64
-{
+fn get_max_value(values: Vec<u64>) -> u64 {
     *values.iter().max().unwrap()
```

Comment:
nit: use `unwrap_or_default()`
Then you can skip all your checks for `if pools_size_vector.len() > 0 { ... } else { default }`

Reply:
You need to consume the iteration, `iter` will give you a reference. Use `into_iter`.